### PR TITLE
footer refactor

### PR DESCRIPTION
**Pull request by Pilu84**

### CODE REVIEW

**Review by laszlocph — COMMENTED** [review]
tldr: Really appreciate your proposal. If you make some changes I will be glad to merge it.

---

There are some issues in the change, I believe you choose a defensive style to introduce your changes, but it makes it impossible to merge the change as is.

I really appreciate the React considerations, some techniques I never used and it is good to see how others do stuff. I will learn about those techniques thanks to you.

The project does not have a linter today, I think some things in this PR (4 spaces) should have been caught by the linter. If you want to add a linter, that would be super valuable.

**Review by laszlocph — APPROVED** [review]

**Comment by laszlocph** [comment]
This is top notch!

Thank you for your contribution!
